google/cloud/pubsub/internal/flow_controlled_publisher_tracing_connection.cc

```diff
@@ -37,8 +37,9 @@ class FlowControlledPublisherTracingConnection
   ~FlowControlledPublisherTracingConnection() override = default;
 
   future<StatusOr<std::string>> Publish(PublishParams p) override {
-    auto span = internal::MakeSpan(
-        "pubsub::FlowControlledPublisherConnection::Publish");
+    auto span = internal::MakeSpan("publisher flow control");
+    span->SetAttribute("cloud-cxx.function",
+                       "pubsub::FlowControlledPublisherConnection::Publish");
```

Comment:
👍

Reply:
We should try to set the attributes on the span when it is created, so that the attribute can factor into sampling decisions.

Reply:
Can you sample based on name? I can't imagine people wanting to sample on this attribute, if it is 1:1 with the name.

Reply:
> Can you sample based on name?

Yes.

> I can't imagine ...

I can imagine all sorts of crazy things.

---

I think generally, if we know an attribute at span creation time, we should use it to initialize the span. If it is a distraction from this PR, we can just open an issue to clean up these cases.

Reply:
I opened #12778

Reply:
SGTM